docs/source/essentials/mutations.md

```diff
@@ -0,0 +1,116 @@
+---
+title: Mutation
+---
+
+Now that we have discussed how to use a query to fetch data from a server using Apollo android, let's look into the other aspect of client-server communication - Sending and writing data to the server.
+Just like REST, any request might end up causing some side-effects on the server, but by convention it's suggested that one doesn't use GET requests to modify data. GraphQL is similar - technically any query could be implemented to cause a data write.
+However, it's useful to establish a convention that any operations that cause writes should be sent explicitly via a *mutation*.
+
+Apollo android also handles GraphQL mutations. Mutations are identical to queries in syntax, the only difference being that you use the keyword `mutation` instead of `query` to indicate that the root fields on this query are going to be performing writes to the backend.
+
+```
+mutation UpvotePost($postId: Int!) {
+  upvotePost(postId: $postId) {
+    votes
+  }
+}
+```
+
+GraphQL mutations represent two things in one query string:
+
+1. The mutation field name with arguments, `upvotePost`, which represents the actual operation to be done on the server
+2. The fields you want back from the result of the mutation to update the client: `{ votes }`
+
+The above mutation will upvote a post on the server. The result might be:
+
+```
+{
+  "data": {
+    "upvotePost": {
+      "id": "123",
+      "votes": 5
+    }
+  }
+}
+```
+
+Similar to queries, mutations are represented by instances of generated classes, conforming to the `ApolloMutationCall` interface. Constructor arguments are used to define mutation variables. You pass a mutation object to `ApolloClient#perform(mutation:)` to send the mutation to the server, execute it, and receive typed results:
+
+```java
+apolloClient
+    .mutate(upvotePostMutation)
+    .enqueue(
```

Comment:
From my point of view (outside team) it will be cool to add `refetchQueries` method to snippet as it's quite useful in cases when no subscription is being used.

Reply:
Can you elaborate a little more? I have limited knowledge about `refetchQueries`. Also, the docs in the code say `Sets a list of Query to be re-fetched once this mutation completed.` and nothing regarding subscription you just mentioned.

Reply:
Actually in this context you may miss Queries that will need to be passed to the `refetchQueries` method. So it's best to leave that out.